### PR TITLE
Print messages about ports that can't change at runtime for K8s

### LINKED ISSUE #32882: Changing port numbers at runtime leads to deployment errors when using K8s/OpenShift extensions
### Describe the bug

Something else I saw today is that the port configuration (for [Management](https://github.com/quarkusio/quarkus/blob/6048280a15e2bc25ed611a35c34b5217f5d83b08/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/management/ManagementInterfaceConfiguration.java#L30) and [HTTP](https://github.com/quarkusio/quarkus/blob/6048280a15e2bc25ed611a35c34b5217f5d83b08/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpConfiguration.java#L27) configuration) are runtime properties.

This means that if users change the port number at runtime, the generated manifests for OpenShift / Kubernetes will stop working. 

I would say this is a very major issue, though the fix (changing these properties to be build-time-only properties) implies adding a strong limitation for users (since users won't be able to update the port to use any longer). 

Not sure if we can / or make sense to allow changing these properties if and only if the Kubernetes extension is not present. 

Relates https://github.com/quarkusio/quarkus/issues/32225#issuecomment-1491335957

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Sgitario**
Related to https://github.com/quarkusio/quarkus/issues/33307, task 3. 
Fix https://github.com/quarkusio/quarkus/issues/32882

I didn't add this logic to Knative (because knative is renaming the port to http1) and Kind/Minikube because these are for DEV purposes only.

### CODE REVIEW

**Comment by Sgitario** [comment]
@iocanel @cescoffier I'm not sure if this is what you had in mind for the task 3 of https://github.com/quarkusio/quarkus/issues/33307.
Note that I don't think this message should use the INFO level, but DEBUG.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java [hunk 1]
```diff
@@ -186,20 +192,40 @@ public static Map<String, Port> combinePorts(List<KubernetesPortBuildItem> ports
...
+     */
+    public static void printMessageAboutPortsThatCantChange(String target, List<KubernetesPortBuildItem> ports,
+            PlatformConfiguration config) {
+        Collection<Port> allPorts = combinePorts(ports, config).values();
+        for (Port port : allPorts) {
+            for (Map.Entry<String, PortProperty> portWithConfig : PORTS_WITH_CONFIG.entrySet()) {
+                if (port.getName().equals(portWithConfig.getKey())) {
+                    LOG.info(String.format("The target '%s' is mapping the container to the port '%s' which value is '%d'. "
+                            + "You won't be able to change it at runtime using the property '%s'.",
+                            target, port.getName(), port.getContainerPort(), portWithConfig.getValue().propertyName));
```
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java:223]:
```suggestion
                    LOG.warn(String.format("'%s' manifests are generated with container port '%s' having value '%d'. "
                            + "The above is influenced by property '%s' which may change at runtime. Such change is likely to cause problems as the application and the container configuration will get out of sync.",
                            target, port.getName(), port.getContainerPort(), portWithConfig.getValue().propertyName));
```

**Review by iocanel — CHANGES_REQUESTED** [review]
I found the message a bit misleading, so I suggested something different.

Also, since this `pattern` is something used in others places too, I am wondering if we should have a utility that does the config reading and logging. It would provide a more concise user experience and possibly make our lives easier years from now when we'll have forgotten all about it.

**Review by cescoffier — CHANGES_REQUESTED** [review]
I like @iocanel suggestion.

**Comment by cescoffier** [comment]
I agree about having a utility class with a consistent, wellformed, not misleading message.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java (hunk 1 above)
Comment (Sgitario) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java:223]: I think the message is now too long. Also, using a warning might cause some panic among users that as far as I know, never complained about this.

**Comment by Sgitario** [comment]
I'm -1 with the Ioannis' suggestion because the message is too long (imagine having the management port plus the HTTP port enabled), and I think that using the warning log level will cause some panic to users that as far as I know, never complained about this.

**Comment by Sgitario** [comment]
After speaking with @iocanel , we agreed with changing the message to:
`[INFO] [io.quarkus.kubernetes.deployment.KubernetesDeployer] The 'kubernetes' manifests are generated with the container port 'http' having value '8080'. The app and manifests will get out of sync if the property 'quarkus.http.port' is changed at runtime.`.

PR updated

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java (hunk 1 above)
Comment (cescoffier) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java]: let's use info instead.

**Review by cescoffier — APPROVED** [review]

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java [hunk 2]
```diff
@@ -291,8 +291,7 @@ public List<DecoratorBuildItem> createDecorators(ApplicationInfoBuildItem applic
             result.add(new DecoratorBuildItem(KUBERNETES, new RemovePortFromServiceDecorator(name, MANAGEMENT_PORT_NAME)));
         }
 
-        printMessageAboutPortsThatCantChange(KUBERNETES, ports, config);
-
+        printMessageAboutPortsThatCantChange(name, ports, config);
```
Comment (Sgitario) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/VanillaKubernetesProcessor.java:294]: I think, name should be KUBERNETES

**Review by Sgitario — COMMENTED** [review]
I'm -1 to include these changes as it overcomplicates the logic for very little benefit (printing messages for users that as far as I know, they never complained about it). 

Though, aside of my comments, I don't want to block this change as long as @iocanel and @cescoffier are ok with proceeding.

**Comment by Sgitario** [comment]
@cescoffier can you take another look to this pull request after the changes made by Ioannis please?

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
The CI issues might be related.

**Review by iocanel — APPROVED** [review]
LGTM

**Comment by iocanel** [comment]
Don't think the failures are related.

**Comment by iocanel** [comment]
> Don't think the failures are related.
Verified that the same tests fail in other pull requests too. So, I'll merge.

**Pull request merged**